### ISSUE TITLE
 failed to find any PEM data in certificate input

### ISSUE THREAD

**Issue by EthanAlban** [clastix/capsule#509]
I can still get this err 

1. **[root@localhost ~]# kubectl version -o json**
```

{
  "clientVersion": {
    "major": "1",
    "minor": "21",
    "gitVersion": "v1.21.8",
    "gitCommit": "4a3b558c52eb6995b3c5c1db5e54111bd0645a64",
    "gitTreeState": "clean",
    "buildDate": "2021-12-15T14:52:11Z",
    "goVersion": "go1.16.12",
    "compiler": "gc",
    "platform": "linux/amd64"
  },
  "serverVersion": {
    "major": "1",
    "minor": "21",
    "gitVersion": "v1.21.8",
    "gitCommit": "4a3b558c52eb6995b3c5c1db5e54111bd0645a64",
    "gitTreeState": "clean",
    "buildDate": "2021-12-15T14:46:22Z",
    "goVersion": "go1.16.12",
    "compiler": "gc",
    "platform": "linux/amd64"
  }
}

```
2. **[root@localhost capsule]# ./hack/create-user.sh alice oil**
```
creating certs in TMPDIR /tmp/tmp.kxJMSEZKh0 
merging groups /O=capsule.clastix.io
Generating RSA private key, 2048 bit long modulus
...................+++
....................................................................+++
e is 65537 (0x10001)
certificatesigningrequest.certificates.k8s.io "alice-oil" deleted
certificatesigningrequest.certificates.k8s.io/alice-oil created
certificatesigningrequest.certificates.k8s.io/alice-oil approved
kubeconfig file is: alice-oil.kubeconfig
to use it as alice export KUBECONFIG=alice-oil.kubeconfig
```
[root@localhost capsule]#  export KUBECONFIG=alice-oil.kubeconfig

3. **[root@localhost capsule]# kubectl get pod**
E0131 02:29:13.633905   32682 cert_rotation.go:168] key failed with : tls: failed to find any PEM data in certificate input
E0131 02:29:13.639601   32682 cert_rotation.go:168] key failed with : tls: failed to find any PEM data in certificate input
error: You must be logged in to the server (the server has asked for the client to provide credentials (get pods))

4. **[root@localhost capsule]# kubectl get csr alice-oil -o json**
```
{
    "apiVersion": "certificates.k8s.io/v1",
    "kind": "CertificateSigningRequest",
    "metadata": {
        "annotations": {
            "kubectl.kubernetes.io/last-applied-configuration": "{\"apiVersion\":\"certificates.k8s.io/v1\",\"kind\":\"CertificateSigningRequest\",\"metadata\":{\"annotations\":{},\"name\":\"alice-oil\"},\"spec\":{\"groups\":[\"system:authenticated\"],\"request\":\"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURSBSRVFVRVNULS0tLS0KTUlJQ2NqQ0NBVm9DQVFBd0xURU9NQXdHQTFVRUF3d0ZZV3hwWTJVeEd6QVpCZ05WQkFvTUVtTmhjSE4xYkdVdQpZMnhoYzNScGVDNXBiekNDQVNJd0RRWUpLb1pJaHZjTkFRRUJCUUFEZ2dFUEFEQ0NBUW9DZ2dFQkFNS3ZqN1JKCnpCbWF2eU9hZ1Rsa2owSG56cFlqaHR4a3R4TkVSZ0xjczRNcDVPb2JCYml1MHlVTzNVTnFHYVpsZEVDWTdRb3IKWDJtVTRVVVJXVVhCQ2tzelRITkR6d3BMMVRRT1ljVVpBc0NKYmVRU1NwVFNUMUpVSnk0c2pQbGllRVpEYkJ5TApyQnQvalF4T01LOEJvdnNNWm9PdTh6QVg4TzQzWjJDYk4wbjFkTU4vVVdBRE1mSjIwRzdDdkVTZk10Vmt4bkFFCldlS2VTS2hDVmp3T0FtN0VwaldjRFpocUZvUzFRV1BUeVhWS0FRTFJVK0tZTWoxbENJcWREQ1ZIWVlkcjRGQVcKYkR0NEFtTFpxem9TUzJsbXowbnh6MnNjbG1oMnZ4bzh3M1FTYUJDTG1OWldsZGlVQjZ4a3RUY2ZQRFdwVlVjcwpCOTFIUWFPQXF6UDQxTTBDQXdFQUFhQUFNQTBHQ1NxR1NJYjNEUUVCQ3dVQUE0SUJBUUJOY1ZyS2tvdVlTdzhzCjY4Tjk3ZXMxSGxQSXk1TTVCdzIvM2RMMDFvQzFnWkVRK2NBMzNWcVB2bmdSd1o0NXlLMkxKUVVjeHBTUWdQajUKbVpiVUZOUkVKeWVpbnFhdXBrTHF1NVpOalluSTUvMFRCQ2tXcitmOVFpcm42NWZFTkV2WFF4bnZSWEN6NFdlcgoxeElZQmI2MG5WNkQyOWx4b3RwTGRZQ2VOS0U2QURrMHZGbkk4eU5jSnJSOWRsUFVPMDVPUGcwTzJhSDhaT0NICjh2Y1hFdXBBbXdWclpVQ01yL0pnRktYcTdCT2IrYS9hUEU0R3dINFJiMDV2YnRHakxYVm9mSHVoZVRDYkNoRWUKQ0E5MlYzQ3FUNnJqb2RnSHp6eXBJQ2hRMlpaQmpOMW1vcTVkaW9zTHJkL1ZScGpJaWRhNEVCZlN4ZzFOTE4zNApleSt3Wk94RwotLS0tLUVORCBDRVJUSUZJQ0FURSBSRVFVRVNULS0tLS0K\",\"signerName\":\"kubernetes.io/kube-apiserver-client\",\"usages\":[\"digital signature\",\"key encipherment\",\"client auth\"]}}\n"
        },
        "creationTimestamp": "2022-01-30T15:39:24Z",
        "name": "alice-oil",
        "resourceVersion": "9220",
        "uid": "5b311241-78ba-4e81-8626-a3edb8938cad"
    },
    "spec": {
        "extra": {
            "principalid": [
                "local://user-vmrtf"
            ],
            "username": [
                "admin"
            ]
        },
        "groups": [
            "system:authenticated",
            "system:cattle:authenticated"
        ],
        "request": "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURSBSRVFVRVNULS0tLS0KTUlJQ2NqQ0NBVm9DQVFBd0xURU9NQXdHQTFVRUF3d0ZZV3hwWTJVeEd6QVpCZ05WQkFvTUVtTmhjSE4xYkdVdQpZMnhoYzNScGVDNXBiekNDQVNJd0RRWUpLb1pJaHZjTkFRRUJCUUFEZ2dFUEFEQ0NBUW9DZ2dFQkFNS3ZqN1JKCnpCbWF2eU9hZ1Rsa2owSG56cFlqaHR4a3R4TkVSZ0xjczRNcDVPb2JCYml1MHlVTzNVTnFHYVpsZEVDWTdRb3IKWDJtVTRVVVJXVVhCQ2tzelRITkR6d3BMMVRRT1ljVVpBc0NKYmVRU1NwVFNUMUpVSnk0c2pQbGllRVpEYkJ5TApyQnQvalF4T01LOEJvdnNNWm9PdTh6QVg4TzQzWjJDYk4wbjFkTU4vVVdBRE1mSjIwRzdDdkVTZk10Vmt4bkFFCldlS2VTS2hDVmp3T0FtN0VwaldjRFpocUZvUzFRV1BUeVhWS0FRTFJVK0tZTWoxbENJcWREQ1ZIWVlkcjRGQVcKYkR0NEFtTFpxem9TUzJsbXowbnh6MnNjbG1oMnZ4bzh3M1FTYUJDTG1OWldsZGlVQjZ4a3RUY2ZQRFdwVlVjcwpCOTFIUWFPQXF6UDQxTTBDQXdFQUFhQUFNQTBHQ1NxR1NJYjNEUUVCQ3dVQUE0SUJBUUJOY1ZyS2tvdVlTdzhzCjY4Tjk3ZXMxSGxQSXk1TTVCdzIvM2RMMDFvQzFnWkVRK2NBMzNWcVB2bmdSd1o0NXlLMkxKUVVjeHBTUWdQajUKbVpiVUZOUkVKeWVpbnFhdXBrTHF1NVpOalluSTUvMFRCQ2tXcitmOVFpcm42NWZFTkV2WFF4bnZSWEN6NFdlcgoxeElZQmI2MG5WNkQyOWx4b3RwTGRZQ2VOS0U2QURrMHZGbkk4eU5jSnJSOWRsUFVPMDVPUGcwTzJhSDhaT0NICjh2Y1hFdXBBbXdWclpVQ01yL0pnRktYcTdCT2IrYS9hUEU0R3dINFJiMDV2YnRHakxYVm9mSHVoZVRDYkNoRWUKQ0E5MlYzQ3FUNnJqb2RnSHp6eXBJQ2hRMlpaQmpOMW1vcTVkaW9zTHJkL1ZScGpJaWRhNEVCZlN4ZzFOTE4zNApleSt3Wk94RwotLS0tLUVORCBDRVJUSUZJQ0FURSBSRVFVRVNULS0tLS0K",
        "signerName": "kubernetes.io/kube-apiserver-client",
        "usages": [
            "digital signature",
            "key encipherment",
            "client auth"
        ],
        "username": "user-vmrtf"
    },
    "status": {
        "conditions": [
            {
                "lastTransitionTime": "2022-01-30T15:39:24Z",
                "lastUpdateTime": "2022-01-30T15:39:24Z",
                "message": "This CSR was approved by kubectl certificate approve.",
                "reason": "KubectlApprove",
                "status": "True",
                "type": "Approved"
            }
        ]
    }
}

```

**Comment by prometherion** [comment]
This is interesting, it's something we already faced in another issue (#385).

This time we got more data and what took my attention is the `/spec/groups` field: I think you're using Rancher since the `system:cattle:authenticated` group is used by them and I started to look for issues with `CertificateSigningRequest` and went through this one.

https://github.com/k3s-io/k3s/issues/524

Would be great to understand if your Rancher version is the one affected by this bug.
